packages/react-router/lib/dom/ssr/routeModules.ts

```diff
@@ -18,19 +18,82 @@ export interface RouteModules {
   [routeId: string]: RouteModule | undefined;
 }
 
+/**
+ * The shape of a route module shipped to the client
+ */
 export interface RouteModule {
+  /**
+   * Action function called only in the browser.  Route client loaders provide
+   * data to route components in addition to, or in place of, route loaders.
+   */
   clientAction?: ClientActionFunction;
+  /**
+   * Like route actions but only called in the browser.
+   */
   clientLoader?: ClientLoaderFunction;
-  ErrorBoundary?: ErrorBoundaryComponent;
-  HydrateFallback?: HydrateFallbackComponent;
+  /**
+   * ErrorBoundary to display for this route
+   */
+  ErrorBoundary?: React.ComponentType;
```

Comment:
I got rid of the type aliases here in favor of just using `React.Component` type directly.